### PR TITLE
[KYUUBI #5361] [AUTHZ] Support Drop/Truncate Table Commands for Hudi

### LINKED ISSUE #5361: [Subtask] Support drop/truncate table command
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues) and found no similar issues.

### Describe the subtask

Support update/delete/truncate table command

### Parent issue

#5285

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### _Why are the changes needed?_
To close #5361. Kyuubi authz support hudi drop/repair/truncate table commands

- DropHoodieTableCommand: https://github.com/apache/hudi/blob/master/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/DropHoodieTableCommand.scala
- TruncateHoodieTableCommand: https://github.com/apache/hudi/blob/master/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/TruncateHoodieTableCommand.scala

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_
No

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @bowenliang123 @yaooqinn

**Review by bowenliang123 — APPROVED** [review]
LGTM.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/HudiCatalogRangerSparkExtensionSuite.scala [hunk 1]
```diff
@@ -206,4 +206,77 @@ class HudiCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
...
+             | primaryKey = 'id',
+             | 'hoodie.datasource.hive_sync.enable' = 'false'
+             |)
+             |PARTITIONED BY(city)
+             |""".stripMargin))
+      interceptContains[AccessControlException](
+        doAs(
+          someone,
+          sql(s"DROP TABLE IF EXISTS $namespace1.$table1".stripMargin)))(
+        s"does not have [drop] privilege on [$namespace1/$table1]")
```
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/HudiCatalogRangerSparkExtensionSuite.scala:230]:
```suggestion
      interceptContains[AccessControlException] {
        doAs(someone, sql(s"DROP TABLE IF EXISTS $namespace1.$table1".stripMargin))
      }(s"does not have [drop] privilege on [$namespace1/$table1]")
```
Change to use "{}" embraces for interceptContains. To make the code more compact in style.

**Comment by bowenliang123** [comment]
Please ensure both POSITIVE and NEGATIVE test cases next time. Positive cases have been added.

**Comment by AngersZhuuuu** [comment]
> Please ensure both POSITIVE and NEGATIVE test cases next time. Positive cases have been added.

👌

**Review by yaooqinn — APPROVED** [review]

**Comment by AngersZhuuuu** [comment]
Seems the repair table command generates a v2 table when running UT together, but generates v1 table when running this test separately...checking

**Comment by yaooqinn** [comment]
> Seems the repair table command generates a v2 table when running UT together, but generates v1 table when running this test separately...checking

It's OK to separate it from this PR

**Comment by AngersZhuuuu** [comment]
> > Seems the repair table command generates a v2 table when running UT together, but generates v1 table when running this test separately...checking
> 
> It's OK to separate it from this PR

DOne

**Review by cxzl25 — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**